### PR TITLE
mesa: 21.3.5 -> 21.3.6

### PR DESCRIPTION
**Pull request by primeos**
###### Motivation for this change

Notes: https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/14505 is now merged and backported via https://gitlab.freedesktop.org/mesa/mesa/-/commit/9b50b320d1db19c8c00be0e5476ef58c55db6e49 (required for https://github.com/NixOS/nixpkgs/issues/153377).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by primeos** [comment]
@ofborg test tinywl sway

**Comment by vcunat** [comment]
Note: `master..staging-next` is empty right now.

**Comment by primeos** [comment]
@vcunat sorry, I'm still too unfamiliar with our `staging` process to properly understand all of the implications.
Does that mean I should target `staging` instead (or wait for a PR to merge `staging-next` back into `master` (as `staging` was merged into `staging-next` at that point)) to avoid causing unnecessary rebuilds on `staging-next` (simplified summary)?

**Comment by vcunat** [comment]
TL;DR: I simply retargeted it, as I expect it's to be merged very soon &ndash; before the next cycle starts. (the bump looks ready to me, though I haven't tested it)

Longer explanation:
- merges `master -> staging-next -> staging` happen automatically ever 6 hours (if conflict-less)
- the rest is manual
- actually we have a nice diagram in the manual: https://nixos.org/manual/nixpkgs/unstable/#idm140737315051616

**Comment by primeos** [comment]
@vcunat thank you for the reply. My question was why it's a problem to merge this into `staging-next` instead of `staging`. That part isn't clear to me yet. From the workflow (that part I do know based on the manual and [RFC](https://github.com/NixOS/rfcs/blob/master/rfcs/0026-staging-workflow.md)) it should be no problem (it'll automatically land in `staging` and also manually in `master` after the next `staging-next -> master` PR). So my assumption was that staging is preferred if there's no `staging-next` PR yet as merging it would cause 1001-2500 rebuilds after the next `staging-next` evaluation but all of them will likely have to be repeated for the next `staging-next` PR as `staging` will get merged into `staging-next`. Is that the reason or am I missing something else?

**Comment by vcunat** [comment]
Yes, Hydra would start those useless rebuilds.
